### PR TITLE
gui-apps/wf-recorder: dependency use flag fixes + live ebuild

### PR DESCRIPTION
**Pull request by epsilon-0**
added proper flags on ffmpeg for encoding and opencl
also added live ebuild

Package-Manager: Portage-2.3.99, Repoman-2.3.22
Signed-off-by: Aisha Tammy <gentoo@aisha.cc>

### CODE REVIEW

**Comment by epsilon-0** [comment]
no revision bump as has no effects on people who have managed to build it correctly wrt their use cases.

**Comment by heroin-moose** [comment]
Thanks!

**Comment by heroin-moose** [comment]
@juippis, could you please take a look?

File: gui-apps/wf-recorder/wf-recorder-0.2.1.ebuild [hunk 1]
```diff
@@ -7,20 +7,32 @@ inherit meson
...
 DESCRIPTION="Screen recorder for wlroots-based compositors"
 HOMEPAGE="https://github.com/ammen99/wf-recorder"
-SRC_URI="https://github.com/ammen99/${PN}/releases/download/v${PV}/${P}.tar.xz"
+
+if [[ ${PV} == 9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/ammen99/wf-recorder.git"
+else
+	SRC_URI="https://github.com/ammen99/wf-recorder/releases/download/v${PV}/${P}.tar.xz"
+	KEYWORDS="~amd64 ~arm64 ~x86"
```
Comment (juippis) [inline gui-apps/wf-recorder/wf-recorder-0.2.1.ebuild:16]: Where did the `arm64` come from? You can't add keywords.

File: gui-apps/wf-recorder/wf-recorder-0.2.1.ebuild [hunk 2]
```diff
@@ -7,20 +7,32 @@ inherit meson
...
 	media-sound/pulseaudio
-	media-video/ffmpeg
-	opencl? ( virtual/opencl )"
+	media-video/ffmpeg[opencl?,pulseaudio,x264]
+	opencl? ( virtual/opencl )
+"
 RDEPEND="${DEPEND}"
-BDEPEND="man? ( app-text/scdoc )"
+BDEPEND="
+	${DEPEND}
```
Comment (juippis) [inline gui-apps/wf-recorder/wf-recorder-0.2.1.ebuild:31]: Can you tell me why you think this is needed? `BDEPEND` was designed to be a splitted off from `DEPEND`. There are cases where this makes sense, but I don't believe this to be one.

File: gui-apps/wf-recorder/wf-recorder-0.2.1.ebuild [hunk 3]
```diff
@@ -7,20 +7,32 @@ inherit meson
...
-	dev-libs/wayland-protocols
+DEPEND="
+	dev-libs/wayland
 	media-sound/pulseaudio
-	media-video/ffmpeg
-	opencl? ( virtual/opencl )"
+	media-video/ffmpeg[opencl?,pulseaudio,x264]
+	opencl? ( virtual/opencl )
+"
 RDEPEND="${DEPEND}"
```
Comment (juippis) [inline gui-apps/wf-recorder/wf-recorder-0.2.1.ebuild:29]: These changes require you to revbump.

File: gui-apps/wf-recorder/wf-recorder-9999.ebuild [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# Copyright 2020 Gentoo Authors
```
Comment (juippis) [inline gui-apps/wf-recorder/wf-recorder-9999.ebuild:1]: Adding this file should happen in a separate commit. So 1 PR, 2 commits. But please sync this with requested changes to 0.2.1.

File: gui-apps/wf-recorder/wf-recorder-0.2.1.ebuild (hunk 1 above)
Comment (epsilon-0) [reply gui-apps/wf-recorder/wf-recorder-0.2.1.ebuild:16]: Oh, dang. All of @ammen99's packages are working on arm64 and I generally use them there too.  
I reflexively added it here, forgetting it wasn't present before. Will change it up.

Comment (epsilon-0) [reply gui-apps/wf-recorder/wf-recorder-0.2.1.ebuild]: removed

File: gui-apps/wf-recorder/wf-recorder-0.2.1.ebuild (hunk 2 above)
Comment (epsilon-0) [reply gui-apps/wf-recorder/wf-recorder-0.2.1.ebuild]: removed

File: gui-apps/wf-recorder/wf-recorder-0.2.1.ebuild (hunk 3 above)
Comment (epsilon-0) [reply gui-apps/wf-recorder/wf-recorder-0.2.1.ebuild:29]: done

File: gui-apps/wf-recorder/wf-recorder-9999.ebuild (hunk 4 above)
Comment (epsilon-0) [reply gui-apps/wf-recorder/wf-recorder-9999.ebuild:1]: done

File: gui-apps/wf-recorder/wf-recorder-9999.ebuild [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="MIT"
+SLOT="0"
+IUSE="+man opencl"
+
+DEPEND="
+	dev-libs/wayland
+	media-sound/pulseaudio
```
Comment (juippis) [inline gui-apps/wf-recorder/wf-recorder-9999.ebuild:25]: Checking upstream, pulseaudio support seems to be optional now. Please update -9999 ebuild to reflect that.
https://github.com/ammen99/wf-recorder/commit/a9c0cc14f6dad3e9067db1fbd85f4275c6b0b50e

Comment (epsilon-0) [reply gui-apps/wf-recorder/wf-recorder-9999.ebuild]: done

**Comment by juippis** [comment]
Sorry, this PR is already merged :( please open a new PR once gentoo tree syncs.

**Pull request closed without merging**